File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    bool lo = true;
public:
    bool check(TreeNode* rootl , TreeNode* rootr){
        
        if (!rootl && !rootr) return true;
        if (!rootl || !rootr) return false;
        if ( rootl->val != rootr->val) return false;
        return check(rootl->left, rootr->right) && check(rootl->right, rootr->left) ;
        
    }
    bool isSymmetric(TreeNode* root) {
        return check(root->left , root->right);
    }
};

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        def check(left , right):
            if not left and not right : return True
            if left and not right:return False
            if right and not left: return False
            if left.val != right.val : return False
            return check(left.left , right.right) and check(left.right , right.left)
        
        return check(root.left , root.right)